blockchain/chain.go

```diff
@@ -1309,6 +1412,7 @@ func New(config *Config) (*BlockChain, error) {
 		blocksPerRetarget:   int32(targetTimespan / targetTimePerBlock),
 		index:               newBlockIndex(config.DB, params),
 		hashCache:           config.HashCache,
+		bestChain:           newChainView(nil),
```

Comment:
After looking more closely, I see that `initChainState` is calling `.SetTip` on every loop iteration.  I will move this out of the loop and after the assertion that the tip is correct (also addressing your other comment about using the local) and that will ensure only a single allocation of the right size.  Agree?